docs/spec/spec.md

```diff
@@ -0,0 +1,251 @@
+
+The SAP JCo Connector architecture consists of key components that facilitate communication between Ballerina applications and SAP systems. These components include the SAP function modules, the SAP system itself, Ballerina code that handles the RFC calls, and the IDoc servers responsible for sending and receiving IDocs. This architecture enables robust integration, supporting various enterprise operations.
+
+## 2. Components
+
+### 2.1 RFC Client
+
+#### 2.1.1 Configurations
+
+##### 2.1.1.1 DestinationConfig
+
+The `DestinationConfig` type holds the configuration details needed to create a BAPI connection.
+
+- **ashost**: The SAP host name (`jco.client.ashost`).
+- **sysnr**: The SAP system number (`jco.client.sysnr`).
+- **jcoClient**: The SAP client (`jco.client.client`).
+- **user**: The SAP user name (`jco.client.user`).
+- **passwd**: The SAP password (`jco.client.passwd`).
+- **lang**: The SAP language (`jco.client.lang`). Default is "EN".
+- **group**: The SAP group (`jco.client.group`). Default is "PUBLIC".
+
+**Example Configuration:**
+
+```ballerina
+configurable jco:DestinationConfig config = ?;
+jco:Client sapClient = check new (config);
+```
+
+Config.toml file:
+
+```toml
+[config]
+ashost = "host"
+sysnr = "00"
+jcoClient = "000"
+user = "user"
+passwd = "password"
+lang = "en"
+group = "group"
+```
+
+##### 2.1.1.2 AdvancedConfig
+
+The `AdvancedConfig` type is a map that holds any custom configurations needed to create a SAP connection.
+
+**Example Configuration:**
+
+```ballerina
+configurable jco:AdvancedConfig config = ?;
+jco:Client sapClient = check new (config);
+```
+
+Config.toml file:
+
+```toml
+[config]
+"jco.destination.pool_capacity" = "3"
+"jco.destination.peak_limit" = "10"
+```
+
+#### 2.1.2 Available Operations
+
+##### 2.1.2.1 Execute a Function Module via RFC
+
+You can execute a specific function module on the SAP system using the `execute` function. This operation requires you to specify the function module name and the necessary parameters.
+
+```ballerina
+public function main() returns error? {
+    ImportParams importParams = {
+        importParam1: "Hello",
+        importParam2: 123,
+        importParam3: 123.45,
+        importParam4: 123.456
+    };
+
+    ExportParams? result = check jcoClient->execute("TEST_FUNCTION", importParams);
+    if (result is jco:ExportParams) {
+        io:println("Result: ", result);
+    } else {
+        io:println("Error: Function execution failed");
+    }
+}
+```
+
+##### 2.1.2.2 Send IDocs to an SAP System
+
+The RFC Client also supports sending IDocs to an SAP system, allowing you to automate the exchange of structured data.
+
+```ballerina
+public function main() returns error? {
+    xml iDoc = check io:fileReadXml("resources/sample.xml");
+    check jcoClient->sendIDoc(iDoc);
+    io:println("IDoc sent successfully.");
+}
+```
+
+#### 2.1.3 Data Type Mapping Between Ballerina and SAP JCo
+
+##### 2.1.3.6 Ballerina to SAP JCo Mapping
```

Comment:
Do we need another sub title here

Reply:
Yes, that is unnecessary. Changed with [1a3f2d5](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/56/commits/1a3f2d588a8e26fea96920ea4152397a479440e6).